Battle Royal.playground/Contents.swift

```diff
@@ -0,0 +1,488 @@
+import UIKit
+
+protocol Resource{
+    var name: String { get }
+    
+    var damageBonus: Int { get }
+    var healBonus: Int {get}
+}
+
+protocol Unit{
+    var hp: Int { get set }
+    var damage: Int { get }
+    var luck: Int { get }
+    var haveUniqueSkill: Bool {get set}
+    
+    var resource: Resource? {get set}
+    
+    func hit(player: Player)
+    func findResource()
+    
+    func voiceOfDeath()
+    func voiceOfKilling()
+    func voiceOfHavenotSkill()
+    func voiceOfFind()
+}
+
+class MagicKnife: Resource{
+    var name: String = "MagicKnife"
+    
+    var damageBonus: Int = 10
+    
+    var healBonus: Int = 0
+    
+}
+
+class Medikit: Resource{
+    var name: String = "Medikit"
+    
+    var damageBonus: Int = 0
+    
+    var healBonus: Int = 10
+    
+    
+}
```

Comment:
Стараемся писать без лишних пустых мест:
```suggestion
class Medikit: Resource{
    var name: String = "Medikit"
    var damageBonus: Int = 0
    var healBonus: Int = 10
}
```